src/node_quic_session.cc

```diff
@@ -2125,12 +2145,13 @@ void QuicSession::StreamReset(
 
 void QuicSession::UpdateIdleTimer() {
   CHECK_NOT_NULL(idle_);
-  uint64_t now = uv_hrtime();
-  uint64_t expiry = ngtcp2_conn_get_idle_expiry(Connection());
-  uint64_t timeout = (expiry - now) / 1000000UL;
-  if (expiry < now || timeout == 0) timeout = 1;
-  Debug(this, "Updating idle timeout to %" PRIu64, timeout);
-  idle_->Update(timeout);
+  // uint64_t now = uv_hrtime();
+  // uint64_t expiry = ngtcp2_conn_get_idle_expiry(Connection());
+  // uint64_t timeout = expiry > now ? (expiry - now) / 1000 : 1;
+  // if (timeout == 0) timeout = 1;
+  // Debug(this, "Updating idle timeout to %" PRIu64, timeout);
+  // idle_->Update(timeout);
+  idle_->Update(10000);  // TODO(@jasnell): Fix
```

Comment:
Did this pass tests? (I think I remember this causing a bit of trouble with those…) Or, more generally, is this something that should be addressed before the PR lands?

Reply:
oh, right right. Yes, thank you. With the update to ngtcp2 there's something wonky about the calculation so I had to temporarily disable. Yes, this needs to be fixed before this lands.

Reply:
Interesting... the timeout calculation is working just fine on Linux but not on Windows... hmmmm

Reply:
Ok, confirmed... yeah, this is very weird... On linux, the expiry calculation works as expected, with the expiry time reported by the ngtcp2 API returning a timestamp greater than the current time returned by uv_hrtime()... on windows, however, the expiry time is always less than uv_hrtime() so the timeout is setting as 1..

Reply:
It's happening with DefaultApplication and Http3Application. It's something that has changed directly within ngtcp2.

Reply:
Figured it out. The idle_timeout was being set assuming milliseconds but the code was expecting nanoseconds. It was throwing the calculation off.